### PR TITLE
ref(subscriptions): Decouple subscription logic from datasets [INGEST-627]

### PR DESCRIPTION
**Pull request by ahmedetefy**
Replica of #2244 
Re-opening because I accidentally merged it into the main branch

### CODE REVIEW

File: snuba/web/views.py [hunk 1]
```diff
@@ -435,25 +438,49 @@ def handle_subscription_error(exception: InvalidSubscriptionError) -> Response:
     )
 
 
+@application.route("/<dataset:dataset>/<entity:entity>/subscriptions", methods=["POST"])
 @application.route("/<dataset:dataset>/subscriptions", methods=["POST"])
```
Comment (lynnagara) [inline snuba/web/views.py:442]: Are you planning to remove this one as a follow up? I think we should

File: snuba/web/views.py [hunk 2]
```diff
@@ -435,25 +438,49 @@ def handle_subscription_error(exception: InvalidSubscriptionError) -> Response:
...
     )
 
 
+@application.route(
+    "/<dataset:dataset>/<entity:entity>/subscriptions/<int:partition>/<key>",
+    methods=["DELETE"],
+)
 @application.route(
     "/<dataset:dataset>/subscriptions/<int:partition>/<key>", methods=["DELETE"]
 )
```
Comment (lynnagara) [inline snuba/web/views.py:471]: Same here, let's deprecate this one once it is no longer being used?

**Comment by lynnagara** [comment]
Just some questions about which endpoints we'll be deprecating, otherwise LGTM

File: snuba/web/views.py (hunk 1 above)
Comment (ahmedetefy) [reply snuba/web/views.py:442]: Yep that's the plan, I am planning to remove line 442 and 470 i.e. for creation and deletion as a follow up

Comment (lynnagara) [reply snuba/web/views.py:442]: Sounds good, it could be useful to mark or comment these in code so it's clear what will happen to the old routes

**Review by lynnagara — APPROVED** [review]

**Pull request merged**